### ISSUE TITLE
label-content-name-mismatch should not treat hyphenated labels as matching non-hyphenated content

### ISSUE THREAD

**Issue by dbjorge** [dequelabs/axe-core#4311]
A [recent ACT update](https://github.com/act-rules/act-rules.github.io/pull/2101) added test case `e9bbdbec137223e2973c6d2896050770c84c26e5` (Failure Example 4) to [proposed ACT rule `Visible label is part of accessible name (2ee8b8)`](https://www.w3.org/WAI/standards-guidelines/act/rules/2ee8b8/proposed/). axe-core currently passes this example where the updated ACT rule expects it to fail:

```html
<a href="#" aria-label="non-standard">nonstandard</a>
```

This happens because axe-core is much more forgiving about punctuation than the ACT rule. The ACT rule's expectation wording is:

> For each target element, all [text nodes](https://dom.spec.whatwg.org/#text) in the [visible text content](https://www.w3.org/WAI/standards-guidelines/act/rules/2ee8b8/proposed/#visible-text-content) [match characters](https://www.w3.org/WAI/standards-guidelines/act/rules/2ee8b8/proposed/#matching-characters) and are contained within the [accessible name](https://www.w3.org/WAI/standards-guidelines/act/rules/2ee8b8/proposed/#accessible-name) of this target element, except for characters in the [text nodes](https://dom.spec.whatwg.org/#text) used to express [non-text content](https://www.w3.org/TR/WCAG21/#dfn-non-text-content). Leading and trailing [whitespace](https://www.w3.org/WAI/standards-guidelines/act/rules/2ee8b8/proposed/#whitespace) and difference in case sensitivity should be ignored.

The critical difference is in the interpretation of the "except for characters in the [text nodes](https://dom.spec.whatwg.org/#text) used to express [non-text content](https://www.w3.org/TR/WCAG21/#dfn-non-text-content)" clause. `label-content-name-mismatch` current removes all punctuation from both label and content before performing its comparison, presuming that punctuation always qualifies as "non-text content", but this isn't accurate in the case where punctuation creates a meaningful difference in language (eg, "government monitoring program" vs "government-monitoring program").

We do want to keep removing punctuation in some circumstances; for example, the [2.5.3 understanding doc](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html)'s section on Punctation and capitalization clarifies that "First Name" is an acceptable label for a form field with visual content "First Name:", and that "Next" is an acceptable label for a button with visual content "Next...". Maybe we should strip punctuation only at the beginning and end of strings, instead of throughout?

This issue is linked to a `skipTests` entry in `test/act-rules/visible-label-in-accessible-name-2ee8b8.spec.js`.

**Comment by straker** [comment]
This is no longer a failed example but got moved to an [inapplicable example](https://www.w3.org/WAI/standards-guidelines/act/rules/2ee8b8/proposed/#inapplicable-example-6).

**Comment by straker** [comment]
Going to close as duplicate of https://github.com/dequelabs/axe-core/issues/5203 where we are tracking this specific example.
